cylc/flow/task_job_mgr.py

```diff
@@ -554,8 +560,11 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
                     # write flag so that subsequent manual retrigger will
                     # generate a new job file.
                     itask.local_job_file_path = None
-
                     itask.waiting_on_job_prep = False
+
+                if not job_log_dirs:
+                    continue
```

Comment:
Commenting out this the test passes but:
![image](https://github.com/user-attachments/assets/abe3a5fc-669c-4e40-ae1c-4e1bce31b032)

Might be caught with other tests though..